### DISCUSSION TITLE
Commit-reveal scheme for a hash guessing game

### DISCUSSION THREAD

**Discussion by heytdep** [stellar/sorobanathon#36] [General]
Hey, we've (me and @itsfederico) written and explained a contract that implements the commit-reveal proving scheme applied to a hash guessing game.

Here's the repo: https://github.com/Xycloo/soroban-commit-reveal-contract (the explanation is in the README).

**Comment by achallagundla** [comment]
As always thank you for your awsome submission @heytdep! Will get back to you shortly after review!

**Comment by achallagundla** [comment]
Congratulations! Your contribution to Sorobanathon: First Light has been accepted! 🎉
Your submission qualifies for a reward of 1200 XLM!
As always lmk if you have any questions on discord!
